### ISSUE TITLE
CPU Pinning

### ISSUE THREAD

**Issue by abelpc** [shiyanhui/libcsp#3]
Will there be any support for CPU pinning? I know it may need some more scheduling extensions, but this would be a very useful need for some sort of workloads where locality inside a NUMA node matters.

**Comment by yann-shi** [comment]
Good question.

To a coroutine library, there is two kinds of CPU pinning,
- Bind the thread to CPU.
- Bind the coroutine(in libcsp it's called `process`) to CPU.

Currently libcsp only implements the first one(that's why it's so fast).  Sometimes it may be a bad idea to bind a task process to a CPU since other CPUs may starve. What we can do is to try the best to do it without guarantee. Anyway I'll try to implement the second one.

**Comment by abelpc** [comment]
I would personally be interested indeed in the second one. Let the OS decides about which process starves since this would be the user's choice. However, it would perhaps be best to allow/enable users with ways to query the current scheduling state done by libcsp. That's mostly for debugging concerns.

**Comment by yann-shi** [comment]
Yeah, it's important to add the related debugging tools. Libcsp will support it in the future.
